### PR TITLE
Snake brain

### PR DESCRIPTION
**Pull request by dmarcantonio**
This is a class that will Ingest the game state, make a decision, and return an action!

AND TO IT USES FUNCTION CHAINING TO LOOK LIKE A SNAKE 🐍 

`return new SnakeBrain(gameStateResponse).decide().act();`

### CODE REVIEW

**Comment by dmarcantonio** [comment]
I guess I need to comment out unused variables for now
image

File: src/SnakeBrain.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  /**
+   * Where a whole lot of IF/ELSE is going to happen.
+   * This function will determine what behaviours to act upon depending on game state.
+   *
+   * @returns {SnakeBrain}
+   */
+  public decide(): SnakeBrain {
+    // Logic for start of game.
+    console.log(turn);
```
Comment (DevYves) [inline src/SnakeBrain.ts]: console log

File: src/SnakeBrain.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      this.action = Directions.DOWN;
+    }
+
+    // Default action
+    this.action = Directions.LEFT;
+    return this;
+  }
+
+  /**
+   * This direction this snake will take in it's short but exciting life.
```
Comment (DevYves) [inline src/SnakeBrain.ts]: The direction

**Review by samueljohnwilson — APPROVED** [review]
super smart snek

**Review by DevYves — APPROVED** [review]

**Review by mbaart-echosec — APPROVED** [review]
🐍

**Comment by shaun-mac** [comment]
> 🐍

🐢

**Review by shaun-mac — APPROVED** [review]

**Pull request merged**